src/Core/src/Platform/Gtk/MauiDatePicker.cs

```diff
@@ -1,35 +1,443 @@
+		public DateTime Date
 		{
-			Format = string.Empty;
+			get
+			{
+				return _date;
+			}
+		}
+
+		public DateEventArgs(DateTime date)
+		{
+			_date = date;
+		}
+	}
+
+	public class GrabHelper
+	{
+		private static uint CURRENT_TIME = 0;
+
+		public static void GrabWindow(Window window)
+		{
+			window.GrabFocus();
+
+			Grab.Add(window);
+
+			Gdk.GrabStatus grabbed =
+				Gdk.Pointer.Grab(window.Window, true,
+				Gdk.EventMask.ButtonPressMask
+				| Gdk.EventMask.ButtonReleaseMask
+				| Gdk.EventMask.PointerMotionMask, null, null, CURRENT_TIME);
+
+			if (grabbed == Gdk.GrabStatus.Success)
+			{
+				grabbed = Gdk.Keyboard.Grab(window.Window, true, CURRENT_TIME);
+
+				if (grabbed != Gdk.GrabStatus.Success)
+				{
+					Grab.Remove(window);
+					window.Destroy();
+				}
+			}
+			else
+			{
+				Grab.Remove(window);
+				window.Destroy();
+			}
+		}
+
+		public static void RemoveGrab(Window window)
+		{
+			Grab.Remove(window);
+			Gdk.Pointer.Ungrab(CURRENT_TIME);
+			Gdk.Keyboard.Ungrab(CURRENT_TIME);
+		}
+	}
+
+	public partial class DatePickerWindow : Window
+	{
+		VBox _datebox;
+		RangeCalendar _calendar;
+		public delegate void DateEventHandler(object sender, DateEventArgs args);
+
+		public event DateEventHandler? OnDateTimeChanged;
+
+		public DatePickerWindow()
+			: base(WindowType.Popup)
+		{
+			Title = "DatePicker";
+			TypeHint = Gdk.WindowTypeHint.Desktop;
+			WindowPosition = WindowPosition.Mouse;
+			BorderWidth = 1;
+			Resizable = false;
+			Decorated = false;
+			DestroyWithParent = true;
+			SkipPagerHint = true;
+			SkipTaskbarHint = true;
+
+			_datebox = new VBox();
+			_datebox.Spacing = 6;
+			_datebox.BorderWidth = 3;
+
+			_calendar = new RangeCalendar();
+			_calendar.CanFocus = true;
+			_calendar.DisplayOptions = CalendarDisplayOptions.ShowHeading;
+			_datebox.Add(_calendar);
+			Box.BoxChild dateBoxChild = ((Box.BoxChild)(_datebox[_calendar]));
+			dateBoxChild.Position = 0;
+
+			Add(_datebox);
+
+			if ((Child != null))
+			{
+				Child.ShowAll();
+			}
+
+			Show();
+			ButtonPressEvent += new ButtonPressEventHandler(OnButtonPressEvent);
+			_calendar.ButtonPressEvent += new ButtonPressEventHandler(OnCalendarButtonPressEvent);
+			_calendar.DaySelected += new EventHandler(OnCalendarDaySelected);
+			_calendar.DaySelectedDoubleClick += new EventHandler(OnCalendarDaySelectedDoubleClick);
+			GrabHelper.GrabWindow(this);
+			SelectedDate = DateTime.Now;
+		}
+
+		public DateTime SelectedDate
+		{
+			get
+			{
+				return _calendar.Date;
+			}
+
+			set
+			{
+				_calendar.Date = value;
+			}
+		}
+
+		public DateTime MinimumDate
+		{
+			get
+			{
+				return _calendar.MinimumDate;
+			}
+
+			set
+			{
+				_calendar.MinimumDate = value;
+			}
+		}
+
+		public DateTime MaximumDate
+		{
+			get
+			{
+				return _calendar.MaximumDate;
+			}
+
+			set
+			{
+				_calendar.MaximumDate = value;
+			}
+		}
+
+		protected virtual void OnButtonPressEvent(object o, ButtonPressEventArgs args)
```

Comment:
@parhamsaremi please don't use 1-letter var names

Reply:
Most of this code is based on Xamarin, including the line you mentioned above so that's why we didn't change them. should I still fix this?

https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Platform.GTK/Controls/DatePicker.cs#L93

Reply:
no harm in fixing it, most of the other methods use `object sender`

Reply:
sure :+1: